src/Dhall/Freeze.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Dhall.Freeze (freeze) where
+
+import Dhall.Core
+import Dhall.Import (load, hashExpression)
+import Dhall.Parser (exprAndHeaderFromText, Src)
+import Dhall.Pretty (annToAnsiStyle)
+
+import System.Console.ANSI (hSupportsANSI)
+import Data.Monoid ((<>))
+import Data.Maybe (fromMaybe)
+import Data.Text
+
+import qualified Data.Text.Prettyprint.Doc                 as Pretty
+import qualified Data.Text.Prettyprint.Doc.Render.Terminal as Pretty
+import qualified Control.Exception
+import qualified Data.Text.IO
+import qualified System.IO
+
+opts :: Pretty.LayoutOptions
+opts =
+    Pretty.defaultLayoutOptions
+        { Pretty.layoutPageWidth = Pretty.AvailablePerLine 80 1.0 }
+
+readInput :: Maybe FilePath -> IO Text
+readInput = maybe fromStdin Data.Text.IO.readFile
+    where 
+        fromStdin = System.IO.hSetEncoding System.IO.stdin System.IO.utf8 >> Data.Text.IO.getContents
+
+hashImport :: Import -> IO Import
```

Comment:
I think you should also export this utility since it is handy for people using the Haskell API

Reply:
Done.